docs/docs/application-basics.md

```diff
@@ -74,21 +89,78 @@ And how this looks in practice shows you the following example: let's assume you
     "cpus": 0.1,
     "mem": 10.0,
     "instances": 1,
-    "uris": [
-        "https://example.com/app.zip"
+    "fetch": [
+        {
+          "uri": "https://example.com/app.zip"
+        }
     ]
 }
 ```
 
 Note that in contrast to the example `basic-1` we now have a `cmd` that looks as follows: `app/cool-script.sh`. This stems from the fact that when the zip file gets downloaded and extracted, a directory `app` according to the file name `app.zip` is created where the content of the zip file is extracted into.
 
-Note also that you can specify many resources, not only one. So, for example, you could provide a git repository and some resources from a CDN as follows:
+Additionally, if you want to use a fetcher cache, `"cache": true` field must be specified. If a URI is encountered 
+for the first time (for the same user), it is first downloaded into the cache, then copied to the sandbox directory 
+from there. If the same URI is encountered again, and a corresponding cache file is resident in the cache or 
+still en route into the cache, then downloading is omitted and the fetcher proceeds directly to copying from the cache. 
```

Comment:
I think it's worth mentioning that the cache is "per agent" and so effectively if the task is restarted on the new agent each time, the file will be downloaded each time.